src/infra/sslQueryRepo.go

```diff
@@ -0,0 +1,220 @@
+	if len(vhostsMatch) < 1 {
+		return []valueObject.Fqdn{}, err
+	}
+
+	httpdVhosts := []valueObject.Fqdn{}
+	for _, vhostMatchStr := range vhostsMatch {
+		vhostStr := vhostMatchStr[1]
+		vhost, err := valueObject.NewFqdn(vhostStr)
+		if err != nil {
+			log.Printf("UnableToGetVhost (%v): %v", vhostStr, err)
+			continue
+		}
+		httpdVhosts = append(httpdVhosts, vhost)
+	}
+
+	return httpdVhosts, nil
+}
+
+func (repo SslQueryRepo) SslFactory(
+	sslHostname string,
+	sslPrivateKeyStr string,
+	sslCertStr string,
+) (entity.SslPair, error) {
+	var ssl entity.SslPair
+
+	hostname, err := valueObject.NewFqdn(sslHostname)
+	if err != nil {
+		return ssl, errors.New("SslHostnameError")
+	}
+	_, err = repo.GetVhostConfigFilePath(hostname)
+	if err != nil {
+		return ssl, err
+	}
+
+	privateKey, err := valueObject.NewSslPrivateKey(sslPrivateKeyStr)
+	if err != nil {
+		return ssl, err
+	}
+
+	sslCertificates, err := repo.splitSslCertificate(sslCertStr)
+	if err != nil {
+		return ssl, err
+	}
+
+	var certificate entity.SslCertificate
+	var chainCertificates []entity.SslCertificate
+	var chainCertificatesContent []valueObject.SslCertificateStr
+	for _, sslCertificate := range sslCertificates {
+		if !sslCertificate.IsCA {
+			certificate = sslCertificate
+			continue
+		}
+
+		chainCertificates = append(chainCertificates, sslCertificate)
+		chainCertificatesContent = append(chainCertificatesContent, sslCertificate.Certificate)
+	}
+
+	hashId, err := valueObject.NewSslHashIdFromSslPair(
+		certificate.Certificate,
+		chainCertificatesContent,
+		privateKey,
+	)
+	if err != nil {
+		return ssl, err
+	}
+
+	return entity.NewSslPair(
+		hashId,
+		hostname,
+		certificate,
+		privateKey,
+		chainCertificates,
+	), nil
+}
+
+func (repo SslQueryRepo) GetVhostConfigFilePath(
+	vhost valueObject.Fqdn,
+) (valueObject.UnixFilePath, error) {
+	var vhostConfigFilePath valueObject.UnixFilePath
+	httpdContent, err := infraHelper.ReadFile(olsHttpdConfigPath)
+	if err != nil {
+		return "", err
+	}
+
+	vhostConfigFileExpression := "\\s*configFile\\s*(.*)"
+	vhostConfigFileMatch, err := infraHelper.GetRegexUniqueMatch(httpdContent, vhostConfigFileExpression)
+	if err != nil {
+		return "", err
+	}
+
+	vhostConfigFilePath, err = valueObject.NewUnixFilePath(vhostConfigFileMatch)
+	if err != nil {
+		return "", err
+	}
+
+	return vhostConfigFilePath, nil
+}
+
+func (repo SslQueryRepo) GetSslPairs() ([]entity.SslPair, error) {
+	var sslPairs []entity.SslPair
+	httpdVhosts, err := repo.GetVhosts()
+	if err != nil {
+		return []entity.SslPair{}, err
+	}
+
+	for _, vhost := range httpdVhosts {
+		vhostConfigFilePath, err := repo.GetVhostConfigFilePath(vhost)
+		if err != nil {
+			return []entity.SslPair{}, err
+		}
+
+		vhostConfigContentStr, err := infraHelper.ReadFile(vhostConfigFilePath.String())
+		if err != nil {
+			return []entity.SslPair{}, err
+		}
+
+		/* TODO: Remover quando implementar o middleware de validação de serviço. */
```

Comment:
That won't fly.

Reply:
Comentário removido.

Reply:
Ajustado. Arquivo `outdated`. Aguardando revisão.